### ISSUE TITLE
Sub categories show all posts in category?

### ISSUE THREAD

**Issue by tedserbinski** [jekyll/jekyll#2645]
Hi! I'm not sure if this is fully supported or not (I can't find anything that says either way) but the behavior is a bit odd.

Folder structure:
```
_posts/
  category1/
    post0.markdown
    subcategory1/
      post1.markdown
    subcategory2/
      post2.markdown
  category2/
    post3.markdown
  category3/
    subcategory3/
      post4.markdown
```

When I list ```site.categories``` I have an array of all posts for each category and subcategory -- good!

However, each subcategory has all posts inclusive of the parent category -- bad! E.g., if I list subcategory1 it will have all posts under category1, including subcategory1 and subcategory2.

I'm using 2.2.0

**Comment by ickc** [comment]
I think the folder structure @tedserbinski used is currently unsupported. See #4501.

**Comment by ickc** [comment]
Should this issue be closed for the reason explained in #4501?

**Comment by parkr** [comment]
@ickc Yes, thank you!
